pkg/types/types.go

```diff
@@ -30,19 +33,40 @@ const (
 	KubeEndPoint = "kubelet.sock"
 )
 
-// ResourceConfig contains cofiguration paremeters for a resource pool
+// DeviceType is custom type to define supported device types
+type DeviceType string
+
+const (
+	// NetDeviceType is DeviceType for network class devices
+	NetDeviceType DeviceType = "netDevice"
+	// // AcceleratorType is DeviceType for accelerator class devices
+	// AcceleratorType DeviceType = "accelerator"
+)
+
+// SupportedDevices is map of 'device identifier as string' to 'device class hexcode as int'
+var SupportedDevices = map[DeviceType]int{
+	NetDeviceType: 0x02,
+	// AcceleratorType: 0x12,
+}
+
+// ResourceConfig contains configuration paremeters for a resource pool
 type ResourceConfig struct {
-	ResourcePrefix string `json:"resourcePrefix,omitempty"` // optional resource prefix that will ovewrite global prefix specified in cli params
-	ResourceName   string `json:"resourceName"`             // the resource name will be added with resource prefix in K8s api
-	IsRdma         bool   // the resource support rdma
-	Selectors      struct {
-		Vendors     []string `json:"vendors,omitempty"`
-		Devices     []string `json:"devices,omitempty"`
-		Drivers     []string `json:"drivers,omitempty"`
-		PfNames     []string `json:"pfNames,omitempty"`
-		LinkTypes   []string `json:"linkTypes,omitempty"`
-		DDPProfiles []string `json:"ddpProfiles,omitempty"`
-	} `json:"selectors,omitempty"` // Whether devices have SRIOV virtual function capabilities or not
+	ResourcePrefix string           `json:"resourcePrefix,omitempty"` // optional resource prefix that will ovewrite global prefix specified in cli params
+	ResourceName   string           `json:"resourceName"`             // the resource name will be added with resource prefix in K8s api
+	IsRdma         bool             // the resource support rdma
```

Comment:
We could move `isRdma` into `NetDeviceSelectors` but that will result in compatibility issue with previous versions. Are we OK with this approach?

Reply:
I think that in the issue discussion backward compatibility was desired.
it eventually depends on who deploys this plugin and how.

e.g for Openshift they would need to change their SR-IOV network Operator code and whether or not there is coupling between the Operator's version and the device plugin image version (i.e older Operator pull in a newer sriov device plugin image).

@zshi-redhat thoughts on the matter?

Reply:
If we all agree that `isRdma` is a network specific flag for all rdma devices( including infiniband ),  we shall move it in network specific structure. I think this will provide an example of adding device type sepcific flag.

Re backward compatibility, I was thinking before merging this PR, we will need to create a new release (branch) so that there will be a pinned version for operator to integrate.

Reply:
This was addressed in latest commits that moving `isRdma` flag to `NetDeviceSelectors`.

Note: the change break backward compability, doc update is needed.